DataFormats/ProtonReco/interface/ProtonTrack.h

```diff
@@ -66,20 +128,37 @@ namespace reco
         return LHCSector::invalid;
       }
 
-      // convenience getters for the extra information
-      bool validFit() const { return pt_extra_->validFit(); }
-      ProtonTrackExtra::ReconstructionMethod method() const { return pt_extra_->method(); }
-      const ProtonTrackExtra::CTPPSLocalTrackLiteRefVector& contributingLocalTracks() const { return pt_extra_->contributingLocalTracks(); }
-
-      void setProtonTrackExtra( const ProtonTrackExtraRef& ref ) { pt_extra_ = ref; }
-      const ProtonTrackExtraRef& protonTrackExtra() const { return pt_extra_; }
-
     private:
-      float xi_; ///< fractional momentum loss (positive for diffractive protons)
-      ProtonTrackExtraRef pt_extra_; ///< Additional information on proton track
-
       static constexpr float mass_ = 0.938272046; ///< proton mass, GeV
       static constexpr float massSquared_ = 0.88035443; ///< proton mass squared, GeV^2
+
+      /// return the uncertainty on a given component
+      double error( Index i ) const {
+        return sqrt( covariance_( (unsigned int)i, (unsigned int)i ) );
+      }
+
+      /// reconstructed vertex position at z/s = 0
+      Point vertex_;
+      /// reconstructed momentum vector
+      Vector momentum_;
+      /// reconstructed time at forward detectors
+      float t_;
+      /// uncertainty on reconstructed time at forward detectors
+      float t_err_;
+      /// fractional momentum loss (positive for diffractive protons)
+      float xi_;
+      /// 5x5 covariance matrix
+      CovarianceMatrix covariance_;
+      /// chi-squared
+      float chi2_;
+      /// number of degrees of freedom
+      unsigned int ndof_;
```

Comment:
Actually, not sure if a `double`/`float` (as used in `reco::Track`) would be more convenient?

Reply:
I can't see how ``ndof`` could become non-integer - any idea?

Reply:
Me neither... but looking at https://github.com/cms-sw/cmssw/blob/master/DataFormats/Candidate/interface/VertexCompositeCandidate.h#L66, https://github.com/cms-sw/cmssw/blob/master/DataFormats/TrackReco/interface/TrackBase.h#L456, ... I am also confused of this "common usage".

Reply:
Personally, I'd go with with integer.

Reply:
the common use was unsigned value a while back, but then it was changed to double/float in 
https://github.com/cms-sw/cmssw/commit/0a7b1e5136be78b5e0806083b7326f7514447fda
This was before I started following the software developments
@llista may know